src/bin/nat.rs

```diff
@@ -0,0 +1,183 @@
+use egg::*;
+use rand::Rng;
+use ruler::{HashMap, *};
+
+define_language! {
+    pub enum Nat {
+        "Z" = Z,
+        "S" = S(Id),
+        "+" = Add([Id; 2]),
+        "*" = Mul([Id; 2]),
+        Var(egg::Symbol),
+    }
+}
+
+impl std::fmt::Display for Type {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "")
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum Type {
+    Nat,
+}
+
+impl Nat {
```

Comment:
can we add some comments here describing why we need to pass the egraph to mk_constant? Might be confusing for the other domains. Maybe we can also add a 1-liner in every domain clarifying that this is needed for the Nat domain..